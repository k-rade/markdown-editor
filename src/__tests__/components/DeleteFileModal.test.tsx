import { fireEvent, render, RenderResult } from "@testing-library/react";
import DeleteFileModal from "../../components/DeleteFileModal";

describe("DeleteFileModal", (): void => {
  const mockOnClose = jest.fn();
  const mockOnSubmit = jest.fn();
  const renderTestComponent = (isOpen: boolean): RenderResult =>
    render(
      <DeleteFileModal
        isOpen={isOpen}
        onClose={mockOnClose}
        onSubmit={mockOnSubmit}
      />
    );

  it("should render", (): void => {
    const { baseElement } = renderTestComponent(true);

    expect(baseElement).toMatchSnapshot();
  });

  it("should not render when is not open", (): void => {
    const { baseElement } = renderTestComponent(false);

    expect(baseElement).toMatchSnapshot();
  });

  it("should close modal when clicked on x icon", (): void => {
    const { getByTestId } = renderTestComponent(true);

    fireEvent.click(getByTestId("x-icon"));

    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });

  it("should close modal when clicked on modal overlay", (): void => {
    const { getByTestId } = renderTestComponent(true);

    fireEvent.click(getByTestId("delete-file-modal-overlay"));

    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });

  it("should close modal when clicked on cancel button", (): void => {
    const { getByText } = renderTestComponent(true);

    fireEvent.click(getByText("Cancel"));

    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });

  it("should confirm deletion when clicked on delete button", (): void => {
    const { getByText } = renderTestComponent(true);

    fireEvent.click(getByText("Delete"));

    expect(mockOnSubmit).toHaveBeenCalledTimes(1);
  });
});
