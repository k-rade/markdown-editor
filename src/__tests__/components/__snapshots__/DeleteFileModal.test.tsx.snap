// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteFileModal should not render when is not open 1`] = `
<body>
  <div />
</body>
`;

exports[`DeleteFileModal should render 1`] = `
<body>
  <div>
    <div
      class="fixed inset-0 bg-zinc-500/50 dark:bg-zinc-900"
      data-testid="delete-file-modal-overlay"
    />
    <div
      class="top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 fixed bg-white shadow-lg p-4 rounded-md w-full max-w-lg dark:bg-zinc-800"
      data-testid="delete-file-modal"
    >
      <div
        class="flex flex-col justify-center items-center gap-4 mb-8 sm:flex-row sm:items-start"
      >
        <div
          class="p-3 bg-rose-100 rounded-full dark:bg-rose-900 dark:bg-opacity-30"
        >
          <svg
            class="stroke-rose-500"
            fill="none"
            height="1.5rem"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1.5rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
            />
            <line
              x1="12"
              x2="12"
              y1="9"
              y2="13"
            />
            <line
              x1="12"
              x2="12.01"
              y1="17"
              y2="17"
            />
          </svg>
        </div>
        <div
          class="flex-1 text-center sm:text-left"
        >
          <h5
            class="flex items-start justify-center font-semibold sm:justify-between dark:text-zinc-100"
          >
            Delete note
            <svg
              class="hidden stroke-zinc-400 cursor-pointer sm:block"
              data-testid="x-icon"
              fill="none"
              height="1.5rem"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="18"
                x2="6"
                y1="6"
                y2="18"
              />
              <line
                x1="6"
                x2="18"
                y1="6"
                y2="18"
              />
            </svg>
          </h5>
          <p
            class="text-zinc-500 text-md dark:text-zinc-400"
          >
            Are you sure you want to delete this note?
            <br />
             This action can not be undone.
          </p>
        </div>
      </div>
      <div
        class="flex flex-col gap-4 justify-end sm:flex-row"
      >
        <button
          class="order-2 shadow-sm font-semibold text-sm border bg-zinc-50 border-zinc-300 px-4 py-2 rounded-md text-zinc-900 hover:bg-zinc-100 sm:order-1 dark:shadow-md dark:text-zinc-100 dark:hover:bg-zinc-700 dark:border-zinc-500"
          type="button"
        >
          Cancel
        </button>
        <button
          class="order-1 shadow-sm font-semibold text-sm border border-rose-700 bg-rose-600 px-4 py-2 rounded-md text-white hover:bg-rose-700 hover:border-rose-800 dark:shadow-md dark:bg-rose-800 dark:hover:bg-rose-900"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</body>
`;
